internal/pkg/deploy/cloudformation/lb_fargate_app.go

```diff
@@ -28,25 +49,60 @@ func NewLBFargateStack(in *deploy.CreateLBFargateAppInput) *LBFargateStackConfig
 
 // StackName returns the name of the stack.
 func (c *LBFargateStackConfig) StackName() string {
-	return ""
+	const maxLen = 128
+	n := fmt.Sprintf("%s-%s-%s-app", c.Env.Project, c.Env.Name, c.App.Name)
+
+	if len(n) > maxLen {
+		return n[len(n)-maxLen:]
+	}
+	return n
 }
 
 // Template returns the CloudFormation template for the application parametrized for the environment.
-func (c *LBFargateStackConfig) Template() string {
-	return ""
+func (c *LBFargateStackConfig) Template() (string, error) {
+	content, err := c.box.FindString(lbFargateAppTemplatePath)
+	if err != nil {
+		return "", &ErrTemplateNotFound{templateLocation: lbFargateAppTemplatePath, parentErr: err}
+	}
+	tpl, err := template.New("template").Parse(content)
+	if err != nil {
+		return "", fmt.Errorf("parse CloudFormation template for %s: %w", c.App.Type, err)
```

Comment:
Er, the verb tense here is confusing to me. Would "Error parsing CloudFormation template..." make more sense?

Reply:
We're trying to follow the recommendation here for the context test: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

> When adding context to returned errors, keep the context succinct by avoiding phrases like "failed to", which state the obvious and pile up as the error percolates up through the stack

Otherwise we can repeat "Error ..." in multiple errors up the call chain, but I agree maybe there is a nicer way of surfacing errors to users. 

I'm not sure how to handle that yet :|